plugins/CoreHome/angularjs/common/services/periods.js

```diff
@@ -348,7 +348,8 @@
         // apply piwik site timezone (if it exists)
         date.setHours((piwik.timezoneOffset || 0) / 3600);
 
-        // get rid of minutes/seconds/etc.
+        // get rid of hours/minutes/seconds/etc.
+        date.setHours(0);
```

Comment:
By default all dates returned for a single day have the time set to `0`. But for today the timezone adjustments may end up on the day (before or after) at a specific hour. The current day isn't highlighted for `today` later, as comparing doesn't match (https://github.com/matomo-org/matomo/blob/3.x-dev/plugins/CoreHome/angularjs/date-picker/date-picker.directive.js#L312-L313) Setting the hour to `0` always should fix that

Reply:
This undoes the setHours() call above, maybe it can be removed? I added it to match the way periods were created server side, but I suppose client side it's just about selecting dates not determining when its needed to aggregate from...

Reply:
Actually it does not really "undo" the set Hours. If the timezone offset is negativ the first `setHours` will move the date to the day before and the `setHours(0)` will then set the hours to `0` on that day. eg:
![image](https://user-images.githubusercontent.com/1579355/44938869-81828080-ad81-11e8-9140-fd0281b853aa.png)

Reply:
Oh I understand now. Looks good, will merge!